### PR TITLE
tests: drivers: gpio: gpio_basic_api: add some STM32 boards overlay

### PR DESCRIPTION
**Pull request by ABOSTM**
tests: drivers: gpio: gpio_basic_api: add some STM32 boards overlay

Add overlay for boards:
* nucleo_f429zi
* nucleo_g474re
* nucleo_l152re
* nucleo_l4r5zi
* stm32f3_disco
* nucleo_f091rc
* nucleo_l073rz

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

### CODE REVIEW

**Comment by ABOSTM** [comment]
Start buildkite error: restart buildkite

**Comment by ABOSTM** [comment]
buildkite error : EDT pickle issue

**Review by erwango — APPROVED** [review]

**Review by pabigot — COMMENTED** [review]
I don't object to this[*], but is it necessary?  The original goal was to have the boards that are used by the GPIO subsystem maintainers to verify that the driver is correct.  So we need at least one for each driver implementation.

I'm not sure there's a value in having one for every board Zephyr supports.  It increases maintenance burden if we have to update the way the hardware configuration is supplied to the test.

@mnkp your call as area maintainer.

[*] well, obviously I do, but not to the point of blocking it

**Comment by ABOSTM** [comment]
The boards I added are the one we setup on STMicroelectronics test bench,
and I need those updates to be able to run this test on the corresponding boards.
Having those tests running on test bench allows to execute tests automatically
... almost every day (not only gpio but the same for other drivers).
Aim is to be able to detect regression on STM32 boards as soon as possible.
And result are push on zephyr tests results server, for example here: 
https://testing.zephyrproject.org/daily_tests/v2.3.0-rc2/report/nucleo_l152re.xml.html

Please note that on test bench, we didn't setup all STM32 boards,
but we try to have 1 per STM32 series, as IPs/drivers may vary from  one series to the other (talking about drivers in general)

**Review by pabigot — APPROVED** [review]
OK, if these support vendor testing without out-of-tree modifications its good.

**Comment by erwango** [comment]
bk failed duer to EDT pickle issue.

**Comment by pabigot** [comment]
> bk failed duer to EDT pickle issue.

Please rebase and push a new version to give buildkite another chance.

**Comment by pabigot** [comment]
I believe you really do need to rebase, so the in-tree buildkite configuration is updated to fix the problem it keeps failing on.

**Pull request merged**
